### PR TITLE
chore: Adding optional AdyenDelegatedAuthentication target

### PR DESCRIPTION
**Pull request by mohammedDehairy**
<new>

## New
- The [Authentication SDK](https://github.com/Adyen/adyen-authentication-ios) for [delegated authentication](https://www.adyen.com/knowledge-hub/delegated-authentication) is now included as an optional module with a version control. To enable it, use your dependency manager to include it in your project. 

</new>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,6 +28,7 @@ DerivedData
 
 # Carthage
 Carthage/
+Cartfile.resolved
```
Comment (descorp) [inline .gitignore:31]: It is needed for Carthage distribution to constrain 3DS and other dependencies.

File: AdyenDelegatedAuthentication/AdyenDelegatedAuthentication.swift [hunk 2]
```diff
@@ -0,0 +1,13 @@
+//
+//  File.swift
+//  
+//
+//  Created by Mohamed Eldoheiri on 14/11/2023.
+//
```
Comment (descorp) [inline AdyenDelegatedAuthentication/AdyenDelegatedAuthentication.swift:6]: Header :)

File: AdyenActions/Components/3DS2/Action handlers/3DS2+Delegated Authentication/ThreeDS2PlusDACoreActionHandler.swift [hunk 3]
```diff
@@ -124,7 +124,7 @@
                 completion(.failure(DelegateAuthenticationError.authenticationFailed(cause: nil)))
                 return
             }
-            delegatedAuthenticationService.authenticate(withBase64URLString: delegatedAuthenticationInput) { result in
+            delegatedAuthenticationService.authenticate(withAuthenticationInput: delegatedAuthenticationInput) { result in
```
Comment (descorp) [inline AdyenActions/Components/3DS2/Action handlers/3DS2+Delegated Authentication/ThreeDS2PlusDACoreActionHandler.swift:127]: This naming is so Obj-c 😅

**Review by descorp — APPROVED** [review]

File: AdyenDelegatedAuthentication/AdyenDelegatedAuthentication.swift (hunk 2 above)
Comment (mohammedDehairy) [reply AdyenDelegatedAuthentication/AdyenDelegatedAuthentication.swift:6]: there're quite a bit of those in the project actually, maybe it should be a separate PR to fix those :-)

**Pull request merged**